test/test_quantize_script.py

```diff
@@ -71,5 +72,101 @@ def forward(self, x):
                    .check_not('Observer = prim::GetAttr[name="_observer_') \
                    .run(m.graph)
 
+
+    def test_insert_quant_dequant_dynamic(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+                self.conv = torch.nn.Conv2d(3, 5, 3).float()
+
+            def forward(self, x):
+                return self.conv(x)
+
+        m = torch.jit.script(M())
+        qconfig = default_dynamic_qconfig
+        qconfig_dict = {
+            '': script_qconfig(qconfig)
+        }
+        m = wrap_cpp_module(torch._C._jit_pass_insert_observers(m._c, "forward", qconfig_dict, False, True))
+        data = torch.randn(1, 3, 10, 10, dtype=torch.float)
+
+        get_forward(m._c)(data)
+
+        m = wrap_cpp_module(torch._C._jit_pass_insert_quant_dequant(m._c, "forward", False, True))
+
+        assert len(m._modules._c.items()) == 1, \
+            'Expected to have single submodule of conv'
+
+        get_forward(m._c)(data)
+        quant_func = "aten::quantize_per_tensor"
+
+        # quantizing activations
+        FileCheck().check("aten::_choose_qparams_per_tensor") \
+                   .check(quant_func) \
+                   .check("prim::CallMethod[name=\"forward\"]") \
+                   .check_not(quant_func) \
+                   .check("return") \
+                   .run(str(get_forward_graph(m._c)))
+        # quantizing weight in forward function of conv module, no choose_qparams
+        FileCheck().check_not("aten::_choose_qparams_per_tensor") \
+                   .check(quant_func) \
+                   .check("prim::CallMethod[name=\"_conv_forward\"]") \
+                   .check_not(quant_func) \
+                   .check("return") \
+                   .run(str(get_forward_graph(m.conv._c)))
+        # shouldn't have quant/dequant in _conv_foward function
+        FileCheck().check_not(quant_func) \
+                   .check("aten::conv2d") \
+                   .check_not(quant_func) \
+                   .check("return") \
+                   .run(str(get_module_method(m, 'conv', '_conv_forward').graph))
+
+    def test_insert_quant_dequant_linear_dynamic(self):
```

Comment:
dynamic_linear?

Reply:
The quantized op is called `qlinear_dynamic` so I went with linear_dynamic instead of dynamic_linear.

Reply:
I see, OK